shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -80,7 +79,7 @@
     private static final byte[] LENGTH_PLACEHOLDER = new byte[4];
     private final Function<Long, AppendBatchSizeTracker> appendTracker;
     private final MetricNotifier metricNotifier;
-    private final Map<Map.Entry<String, UUID>, Session> setupSegments = new HashMap<>();
+    private final Map<String, Session> setupSegments = new HashMap<>();
```

Comment:
This is still not deleting the elements that are no longer used, right? Maybe we could replace the hashmap by a `SimpleCache` class in `common`, which is essentially a thread-safe hash map that incorporates eviction logic. For instance, we could discard any entry of this hash map if it has not been used for some time, right?

Reply:
Thanks for the review!
Yes it's still not deleting outdated segments, I suspect the main reason is that writer reconnects to the same segment with different writerId multiple times.
The `SimpleCache` sounds great, I will take a look at that. Thanks!